tests/test_multi_agent.py

```diff
@@ -0,0 +1,327 @@
+        with open("tests/data/restaurants.json") as f:
+            restaurants = loads(f.read())
+        with open("tests/data/attractions.json") as f:
+            attractions = loads(f.read())
+
+        for city in cities:
+            restaurants_kg.add_node(
+                "City",
+                {
+                    "name": city["name"],
+                    "weather": city["weather"],
+                    "population": city["population"],
+                },
+            )
+            restaurants_kg.add_node("Country", {"name": city["country"]})
+            restaurants_kg.add_edge(
+                "IN_COUNTRY",
+                "City",
+                "Country",
+                {"name": city["name"]},
+                {"name": city["country"]},
+            )
+
+            attractions_kg.add_node(
+                "City",
+                {
+                    "name": city["name"],
+                    "weather": city["weather"],
+                    "population": city["population"],
+                },
+            )
+            attractions_kg.add_node("Country", {"name": city["country"]})
+            attractions_kg.add_edge(
+                "IN_COUNTRY",
+                "City",
+                "Country",
+                {"name": city["name"]},
+                {"name": city["country"]},
+            )
+
+        for restaurant in restaurants:
+            restaurants_kg.add_node(
+                "Restaurant",
+                {
+                    "name": restaurant["name"],
+                    "description": restaurant["description"],
+                    "rating": restaurant["rating"],
+                    "food_type": restaurant["food_type"],
+                },
+            )
+            restaurants_kg.add_edge(
+                "IN_CITY",
+                "Restaurant",
+                "City",
+                {"name": restaurant["name"]},
+                {"name": restaurant["city"]},
+            )
+
+        for attraction in attractions:
+            attractions_kg.add_node(
+                "Attraction",
+                {
+                    "name": attraction["name"],
+                    "description": attraction["description"],
+                    "type": attraction["type"],
+                },
+            )
+            attractions_kg.add_edge(
+                "IN_CITY",
+                "Attraction",
+                "City",
+                {"name": attraction["name"]},
+                {"name": attraction["city"]},
+            )
+
+    def test_multi_agent(self):
+
+        runner = self.orchestrator.ask("Write me a 3 day itinerary for a trip to Italy")
+
+        assert runner is not None
+
+        assert len(runner._agents) == 2, "There should be two agents"
+        assert runner._plan is not None, "Execution plan should not be None"
+
+        response = runner.run()
+
+        assert response is not None
+
+        assert response.text is not None
+
+        assert (
```

Comment:
would you like to validate the existence of a restaurant and an attraction?

Reply:
I think it would be a more complicated validation, but it's a valid change